packages/cmf/__tests__/sagas/documentTitle.test.js

```diff
@@ -0,0 +1,122 @@
+import SagaTester from 'redux-saga-tester';
+import changeDocumentTitle, {
+	formatPath,
+	buildMapFromRoutes,
+	getTitleFromRoutes,
+	assignDocTitle,
+} from '../../src/sagas/documentTitle';
+
+describe('changeDocumentTitle', () => {
+	it('sagas wich initiliaze after receiving events REACT_CMF.REQUEST_SETTINGS_OK and @@router/LOCATION_CHANGE', () => {
+		// given
+		const sagaTester = new SagaTester({ initialState: {} });
+		sagaTester.start(() => changeDocumentTitle());
+		const originalGlobalDoc = global.document;
+		const child2 = { documentTitle: 'child2', path: 'child2' };
+		const child1 = { documentTitle: 'child1', path: 'child1', childRoutes: [child2] };
+		const routes = { documentTitle: 'docTitleRoot', path: '/', childRoutes: [child1] };
+		expect(global.document.title).toBe('');
+		// when
+		sagaTester.dispatch({
+			type: 'REACT_CMF.REQUEST_SETTINGS_OK',
+			settings: { routes },
+		});
+		expect(global.document.title).toBe('docTitleRoot');
+		sagaTester.dispatch({
+			type: '@@router/LOCATION_CHANGE',
+			payload: { pathname: '/child1/child2' },
+		});
+
+		// then
+		expect(global.document.title).toBe('child2');
+		global.document = originalGlobalDoc;
+	});
+});
+
+describe('formatPath', () => {
+	it('format two string to a correct url path', () => {
+		const path = 'world';
+		const parentPath = '/hello';
+		const fPath = formatPath(path, parentPath);
+		expect(fPath).toEqual(`${parentPath}/${path}`);
+	});
+	it('format two string to a correct url path', () => {
+		const path = 'hello';
+		const parentPath = '/';
+		const fPath = formatPath(path, parentPath);
+		expect(fPath).toEqual(`/${path}`);
+	});
+	it('format a string (optional url param) to a correct url path', () => {
+		const path = '(:hello)';
+		const parentPath = '/';
+		const fPath = formatPath(path, parentPath);
+		expect(fPath).toEqual('/:hello?');
+	});
+});
+
+describe('getTitleFromRoutes', () => {
+	it('should return the string myTitle', () => {
+		// Given
+		const location = '/hello/world';
+		const myTitle = 'myTitle';
+		const myDefaultTitle = 'myDefaultTitle';
+		const routes = new Map([[location, myTitle], ['/dumb', 'dumber']]);
+		// When
+		const title = getTitleFromRoutes(routes, location, myDefaultTitle);
+		// Then
+		expect(title).toBe(myTitle);
+	});
+	it('should return the string myDefaultTitle', () => {
+		// Given
+		const location = '/hello/world';
+		const myTitle = 'myTitle';
+		const myDefaultTitle = 'myDefaultTitle';
+		const routes = new Map([['/error', myTitle], ['/dumb', 'dumber']]);
+		// When
+		const title = getTitleFromRoutes(routes, location, myDefaultTitle);
+		// Then
+		expect(title).toBe(myDefaultTitle);
+	});
+});
+
+describe('assignDocTitle', () => {
+	it('should change the document title to Hello World', () => {
```

Comment:
Describe the test case instead of (out of date - here :) ) data
`should change the document title`